datago/src/generator_http.rs

```diff
@@ -0,0 +1,378 @@
+use reqwest::header::HeaderMap;
+use reqwest::header::HeaderValue;
+use reqwest::header::AUTHORIZATION;
+use reqwest::Url;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct SourceDBConfig {
+    pub sources: String,
+    #[serde(default)]
```

Comment:
@DeltaEvo would you know if there's a nicer way than this to auto define defaults ? this makes the struct pretty hard to read, but I couldn't find something much better

Reply:
hmm, maybe that serde connects with the Default trait, if present

Reply:
looks like it's all there is https://serde.rs/attr-default.html, curious to hear your take else I'll try to think of something nicer, for instance get a way to expose the defaults

Reply:
I'm not sure to fully get the usecase, if it's to make the field optional, just wrapping it in an Option would do it

```
pub sources_ne: Option<String>,
```

Reply:
I'd like it to have a default value when a json is deserialized. Option then giving a value when handling the None works I guess, but not as nice as defining it directly in the struct ? it's possible with serde it seems but the value has to come from a callback, which is a quite verbose syntax.. no super big deal

it's like 
#[derive(Deserialize)]
struct MyStruct {
  required_field: String
  optional_field: String = "bar"
}

so that if you deserialize "{"required_field":"foo"}" into this struct, then "optional_field" is defined and takes the value "bar" already. 
I guess another way here is to return the default config explicitly and let the client change what they want

Reply:
Yes, it's either a callback or the type default implementation :/

The neat things with Option is that you get to use all the utility methods on the option type

Reply:
yes, I get it better now, but the issue with Option() is that in this case the default will be spread all over the code wherever I unwrap_or() the values, and I think that's not great design either ? not a big issue, I think I have more important things to improve upon, thanks for all the feedback !